src/community/wps-longitudinal-profile/src/main/java/org/geoserver/wps/longitudinal/LongitudinalProfileProcess.java

```diff
@@ -66,18 +70,45 @@ public class LongitudinalProfileProcess implements GeoServerProcess, DisposableB
 
     private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory();
     private static final String SEP = System.lineSeparator();
+    // for reference, 18k points would currently use 1MB of memory (56 bytes per point)
+    public static final int DEFAULT_MAX_POINTS = 50000;
+    // default chunk size for parallel processing (used to be 1000, load tests shows this 10% faster)
+    public static final int DEFAULT_CHUNK_SIZE = 5000;
 
     private final GeoServer geoServer;
+    private int chunkSize;
+    private int maxPoints;
 
     private ExecutorService executor;
 
     public LongitudinalProfileProcess(GeoServer geoServer) {
         this.geoServer = geoServer;
 
+        this.maxPoints = DEFAULT_MAX_POINTS;
+        try {
+            String maxPointsStr = GeoServerExtensions.getProperty("wpsLongitudinalMaxPoints");
```

Comment:
wpsLongitudinalMaxPoints environment configuration is introduced but there is no documentation addition for it on:
https://docs.geoserver.org/main/en/user/community/wps-longitudinal-profile/index.html

Also we may make it a constant.

This applies also for wpsLongitudinalVerticesChunkSize.